controllers/controllers.go

```diff
@@ -41,19 +41,61 @@ type instanceReconciler struct {
 	recorder record.EventRecorder
 }
 
-// configureInstance adds the specified instance to the cluster. if hostname is not empty, the instance's hostname will be
-// changed to the passed in value. If annotations is not nil, the node will have the specified annotations applied to
-// it.
-func (r *instanceReconciler) configureInstance(instance *instances.InstanceInfo, annotations map[string]string) error {
+// ensureInstanceIsUpToDate ensures that the given instance is configured as a node and upgraded to the specifications
+// defined by the current version of WMCO. If annotations is not nil, the node will have the specified annotations
+// applied to it.
+func (r *instanceReconciler) ensureInstanceIsUpToDate(instance *instances.InstanceInfo, annotationsToApply map[string]string) error {
+	if instance == nil {
+		return errors.New("instance cannot be nil")
+	}
+
+	// Instance is up to date, do nothing
+	if instance.Node != nil && instanceUpToDate(instance.Node.Annotations) {
+		return nil
+	}
+
 	nc, err := nodeconfig.NewNodeConfig(r.k8sclientset, r.clusterServiceCIDR, r.vxlanPort, instance, r.signer,
-		annotations)
+		annotationsToApply)
 	if err != nil {
 		return errors.Wrap(err, "failed to create new nodeconfig")
 	}
-	if err := nc.Configure(); err != nil {
-		return errors.Wrap(err, "failed to configure Windows instance")
+
+	// Check if the instance was configured by a previous version of WMCO and must be deconfigured before being
+	// configured again.
```

Comment:
I ran into an issue where an instance was added to the ConfigMap which resulted in it getting configured. I then removed it from the ConfigMap causing it to be deconfigured. Now when I tried adding it back to the ConfigMap, it failed to get configured because hybrid-overlay failed to come update the new Node object with the annotation because it ignored creating the already existing HNS network. Have you run into this issue?

Reply:
I have not noticed that. Let me double check.